float64[] x
float64[] y
utils/Matrix2D[] covariance